tools/build_langserver/langserver/reference_test.go

```diff
@@ -0,0 +1,83 @@
+package langserver
+
+import (
+	"os"
+
+	"context"
+	"core"
+	"path"
+	"testing"
+
+	"tools/build_langserver/lsp"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetReferences(t *testing.T) {
+	ctx := context.Background()
+
+	// copy over the handler from the setup and get a new analyzer so it would be reading the config
+	a, _ := newAnalyzer()
+	h := handler
+	h.analyzer = a
+
+	uri := lsp.DocumentURI(path.Join(core.RepoRoot, "src/query/BUILD"))
+
+	// TODO(bnm): We need to think of a better way to run handler.getReferences in root directory instead of Chdir here
+	// save the current testing directory
+	testDir, err := os.Getwd()
+	assert.NoError(t, err)
+
+	// Change to repoRoot before we run the test
+	err = os.Chdir(core.RepoRoot)
+	assert.NoError(t, err)
+
+	locs, err := h.getReferences(ctx, uri, lsp.Position{Line: 0, Character: 4})
```

Comment:
I suspect this might be related to the test failures; it isn't really correct to have a test that goes off and walks through the rest of the repo and it might fight with other builds on some targets.